vectorizers/_vectorizers.py

```diff
@@ -390,6 +390,33 @@ def build_skip_grams(
     return new_tokens
 
 
+def build_tree_skip_grams(adjacency_matrix, kernel_function, window_size):
+    """
+    Takes and adjacency matrix counts the co-occurrence of each token within a window_size
+    number of hops from each vertex.  These counts are weighted by the kernel function.
+    Parameters
+    ----------
+    adjacency_matrix: array
+        This should be an adjacency matrix of a graph.
+    kernel_function: a function that takes a window and a window_size parameter
+        and returns a vector of weights.
+    window_size:
+        The size of the window to apply the kernel over.
+
+    Returns
+    -------
+    co_occurences: array of shape adjacency_matrix.shape
+        These are a sparse matrix of weighted co-occurrence counts
+    """
+    weights = kernel_function(np.arange(window_size), window_size)
+    result = adjacency_matrix * weights[0]
+    walk = adjacency_matrix
+    for i in range(1, window_size):
+        walk = walk.dot(adjacency_matrix)
```

Comment:
We're requiring python 3.6+ so you may as well make use of matmul notation.
```suggestion
        walk = walk @ adjacency_matrix
```